src/io/terminal.c

```diff
@@ -76,3 +76,41 @@ void terminal_write(const char* data) {
 		terminal_putchar(data[i]);
 }
 
+// writes the hex representation of a number on the terminal.
+void terminal_write_hex(int n) {
+	if (n < 0) {
+		terminal_putchar('-');
+    	n = -n;
+	}
+
+	// Prefix all hex numbers
+	terminal_write("0x");
+
+	// Can shorten this step when we implement log
+	int num_digits = 0;
+	int tmp = n;
+	do {
+		num_digits++;
+		tmp = tmp / 16;
+	} while (tmp > 0);
+
+	// Get the units of the highest digit.
+	int digitBase = 1;
+	for (int i = 1; i < num_digits; i++) {
+		digitBase = digitBase * 16;
+	}
+
+	// Iteratively extract the digits from highest to lowest.
+	do {
+		char digit = (n / digitBase);
+
+		if (digit < 10) {
+			terminal_putchar('0' + digit);
+		} else {
+			terminal_putchar('A' + digit - 10);
+		}
+
+		n = n % digitBase;
+		digitBase /= 16;
+	} while (n != 0);
```

Comment:
This results in incorrect behavior. We should do `while (num_digits > 0)` and decrement with each iteration.

Reply:
Fixed.